src/Access/ExternalAuthenticators.cpp

```diff
@@ -184,6 +187,17 @@ void parseLDAPServer(LDAPClient::Params & params, const Poco::Util::AbstractConf
     }
     else
         params.port = (params.enable_tls == LDAPClient::Params::TLSEnable::YES ? 636 : 389);
+
+    if (has_search_limit)
+    {
+        const auto search_limit = config.getUInt(ldap_server_config + ".search_limit");
+        // Arbitrary large number.
+        // Previously default value was 100, but in practice one client reported to have 120 roles for a user.
+        if (search_limit >= 2048)
```

Comment:
`search_limit = 0` would mean no limit. So you probably want to check against that too.

However, I don't think there is any real value in introducing another limit on a manually set limit, only possibility of future problems, similar to the one we are fixing here. I thought `100` is plenty...

Reply:
BTW, why do you think that having no limit (i.e. `search_limit = 0`) is not a viable option?

Reply:
I just followed your logic as I understood it. If no more than 2048 entries are allowed, even if the server is attempted to be configured to allow more, then unlimited number of entries should also not be allowed.

However, as I said, this additional limit is strange, and I would allow both 0 and >2048 limit to be configured by the administrator.

Reply:
Removed those checks altogether, whatever integer is set in config is pushed to `params.search_limit`.